model/jpa/src/main/java/org/keycloak/connections/jpa/PersistenceExceptionConverter.java

```diff
@@ -92,11 +96,13 @@ public static ModelException convert(Throwable t) {
         throwModelDuplicateEx = throwModelDuplicateEx.or(checkDuplicationMessage);
 
         if (t.getCause() != null && throwModelDuplicateEx.test(t.getCause())) {
-            throw new ModelDuplicateException(t.getCause());
+            throw new ModelDuplicateException("Duplicate resource error", t.getCause());
         } else if (throwModelDuplicateEx.test(t)) {
-            throw new ModelDuplicateException(t);
+            throw new ModelDuplicateException("Duplicate resource error", t);
+        } else if (t instanceof OptimisticLockException) {
```

Comment:
Does it make sense to have a single `else` branch to throw ` ModelIllegalStateException("Database operation failed", t);` ?

I would say that any other unhandled exception should be thrown as `ModelIllegalStateException`, no?

Reply:
I've now changed that. Now from client perspective they will result in `500` error. Thank you @pedroigor for review!

Reply:
That [last change](https://github.com/keycloak/keycloak/compare/487321bd7c6258e8efc33d38fccf86d7eb566b5c..3930be5d9a67c96d608a500e1801a806bdf98e48) caused failures with mysql and mssql https://github.com/keycloak/keycloak/actions/runs/8634930577/job/23671780092?pr=26096. It seems to me that Keycloak logs are not enabled on those jobs and arranging local test seems bit challenging, so I'm reverting the change for now, unless you'd like me to try going deeper.